code/game/machinery/doors/airlock.dm

```diff
@@ -1016,6 +1044,24 @@ About the new airlock wires panel:
 	if(!forced)
 		if( !hasPower() || isWireCut(AIRLOCK_WIRE_OPEN_DOOR) )
 			return 0
+	if(sabotaged && sabotageTurf)
+		p_open = 1
+		update_icon()
+		visible_message("<span class='warning'>[src]'s panel is blown off in a spray of deadly shrapnel!</span>")
+		explosion(sabotageTurf,-1,1,1)

```

Comment:
what's the point of having two explosions instead of only calling the door charge's ex_act ? (just replace var/sabotaged by var/obj/item/device/doorCharge/sabotaged = X and call X.ex_act)